Adds a try-except block for datasets that may occasionally fail

MuroriM opened the pull request:
- Creates a 'loading_dataset' function wrapper that has a try-except block to catch when the dataset trying to be loaded cannot be reached online.
- Implemented for the MLsum Dataset, which occasionally has this issue

niansong1996 commented on dataset/st_dataset.py:
```diff
@@ -151,4 +151,21 @@ def concatenate_dataset_dicts(dataset_dicts: List[DatasetDict]) -> DatasetDict:
+            if i < tries - 1: # i is zero indexed
+                continue
+            else:
+                sys.exit("Wait for a mninute and attempt downloading the dataset again. The server hosting the dataset occassionally times out.")
```
You could do `raise RuntimeError("message")` and `return None` instead of simply exiting, which is a bit too drastic imo.

niansong1996 commented on dataset/st_dataset.py:
```diff
@@ -151,4 +151,21 @@ def concatenate_dataset_dicts(dataset_dicts: List[DatasetDict]) -> DatasetDict:
+
+def load_dataset_safe(*args, **kwargs) -> Dataset:
+
+    tries = 3
```
you could put this in the function parameter list and have a default value. Remember, having a magic number inside of your code is usually not a good idea ;)

niansong1996 reviewed (commented):
Left some minor comments, otherwise looks great

MuroriM commented on dataset/st_dataset.py:
```diff
@@ -155,17 +155,17 @@ def get_dataset_info(data_dict: DatasetDict) -> DatasetInfo:
                 continue
             else:
-                sys.exit("Wait for a mninute and attempt downloading the dataset again. The server hosting the dataset occassionally times out.")
+                raise RuntimeError("Wait for a mninute and attempt downloading the dataset again. The server hosting the dataset occassionally times out.")
```
@niansong1996 It's unclear how to utilize both 'raise' and 'return' in the same clause so I opted to use 'raise'. Let me know what you think

niansong1996 replied:
Ah, yes. That's a typo from me, you can't do `raise` and `return` in the same block of code. This LGTM

niansong1996 commented:
@MuroriM Lmk when this is done. It seems to me that the passing of #60 actually depends on the changes made here

MuroriM commented:
@niansong1996 I've added the delay between retries.

MuroriM commented:
@MuroriM Sorry, accidentally closed the PR. I'm finished with all the planned changes. Let me know if there's an issue

niansong1996 commented on dataset/st_dataset.py:
```diff
@@ -1,9 +1,11 @@
 
-NUMBER_OF_RETRIES_ALLOWED = 3
+DEFAULT_NUMBER_OF_RETRIES_ALLOWED = 5
+DEFAULT_WAIT_SECONDS_BEFORE_RETRY = 5
```
This is great!

niansong1996 commented on dataset/st_dataset.py:
```diff
@@ -104,30 +104,46 @@ def test_set(self) -> Union[Generator[SummInstance, None, None], List]:
                     continue
                 else:
-                    raise RuntimeError("Wait for a mninute and attempt downloading the dataset again. The server hosting the dataset occassionally times out.")
+                    raise RuntimeError("Wait for a mninute and attempt downloading the dataset again. \
```
You have a typo here for "minute" ;)

niansong1996 reviewed (commented):
Found a typo, looks great otherwise.

I like how you are trying to make docstrings now, good job!

MuroriM commented:
@niansong1996, thanks for the catch. I've fixed the typo

niansong1996 commented:
There is a small conflict which I fixed myself. Should be able to merge after the CI check.

MuroriM commented:
@niansong1996  There was a conflict with the imports between this branch and the notebook branch which has just been merged. I've removed the `sys` module since it's no longer needed and added the `time` and `json` modules to resolve the conflict. Let me know if this is an issue.

MuroriM commented:
@niansong1996 sorry, I hadn't seen your message. I'll be waiting 10 min after your comments before making any changes to avoid this issue in future

niansong1996 commented:
Ah, that's totally okay. Actually I shouldn't have make commits on this branch, my bad. Merging it now.
